ci/ansible/playbooks/os/rhel/upgrade-release.yml

```diff
@@ -0,0 +1,558 @@
+        ansible_control_path: none
+
+    - &REFRESH_ANSIBLE_FACTS
+      name: Refresh Ansible facts
+      setup:
+        gather_subset: min
+
+    - name: Assert latest minor version
+      assert:
+        that: ansible_distribution_version is version(versions.required.full, '>=')
+        quiet: true
+
+    - name: Install packages that provide the leapp utility
+      package:
+        name: "{{ leapp_dependencies.packages[provider] }}"
+        state: present
+
+    - name: Copy leapp metadata archive
+      copy:
+        src: "{{ leapp_archive }}"
+        dest: /etc/leapp/files/leapp-data15.tar.gz
+        mode: preserve
+
+    - name: Unarchive leapp metadata
+      unarchive:
+        src: /etc/leapp/files/leapp-data15.tar.gz
+        dest: /etc/leapp/files/
+        remote_src: true
+
+    # Address upgrade inhibitors
+    # PermitRootLogin option is required in sshd_config file
+    - name: Ensure PermitRootLogin is defined
+      block:
+        - name: Get sshd settings
+          command: sshd -T
+          changed_when: false
+          register: pre_upgrade_sshd_settings
+
+        - name: Check if PermitRootLogin is defined
+          command: grep -i '^PermitRootLogin' /etc/ssh/sshd_config
+          changed_when: false
+          register: grep_permit_root_login
+          failed_when: grep_permit_root_login.rc > 1
+
+        - name: Define PermitRootLogin
+          when:
+            - "'permitrootlogin no' not in pre_upgrade_sshd_settings.stdout_lines"
+            - grep_permit_root_login.rc == 1
+          lineinfile:
+            dest: /etc/ssh/sshd_config
+            regexp: '^#PermitRootLogin'
+            line: 'PermitRootLogin no'
+            state: present
+            backup: true
+          register: sshd_config
+
+    - &RESTART_SSHD
+      name: Restart sshd service
+      when: sshd_config.changed
+      systemd:
+        name: sshd
+        state: restarted
+
+    - name: Disable kernel modules that are not supported in RHEL 8
+      modprobe:
+        name: "{{ item }}"
+        state: absent
+      loop:
+        - floppy
+        - pata_acpi
+
+    - name: Keep only one kernel-devel package when multiple are present
+      block:
+        - name: Get list of all installed packages
+          yum:  # we need envra format thus yum module is used
+            list: installed
+          register: installed_packages
+
+        - name: Register installed versions of kernel-devel package
+          set_fact:
+            installed_kernel_devel_packages: "{{ installed_packages | json_query(_query) }}"
+          vars:
+            _query: results[?name=='kernel-devel'].envra
+
+        - name: Remove old versions of kernel-devel package
+          when: installed_kernel_devel_packages | count > 1
+          yum:
+            name: "{{ installed_kernel_devel_packages[:-1] }}"  # keep the last item
+            state: absent
+
+    ### UPGRADE ###
```

Comment:
Please move this to separate file.

Reply:
Agree that this playbook is too long but it's meant to be used only for our CI pipelines. Having everything in one file allows us to use simple and fast `curl` call to download the playbook in the pipelines. Having multiple files we would have to use more complex curl calls like `curl -O [URL1] -O [URL2]`. Furthermore, I assume that the playbook will be rather short-life code.